airflow/hooks/base_hook.py

```diff
@@ -44,6 +44,21 @@ class BaseHook(LoggingMixin):
     def __init__(self, source):
         pass
 
+    @classmethod
+    def _get_connection_from_db_with_retries(cls, conn_id, max_tries=5):
```

Comment:
wonder if we should encode max_tries into ENV. we're grabbing the connection from ENV anyway? then default to 5 if we can't grab it from ENV.

Reply:
yep we can do that; i'll break out this from the gcs changes first